src/InputSettings.h

```diff
@@ -0,0 +1,12 @@
+#pragma once
+#include "SettingType.h"
+#include<map>
+
+namespace InputSettings {
+    const int MAX_PLAYERS = 4; // Maximum number of players in a game
+    extern int DEADZONE; // Minimum range of displacement for joystick before reading event
+
+    const std::map<std::string, std::pair<const SettingType, void* const>> CONFIG_MAPPING = {
```

Comment:
A map of settings isn't necessary imo, adds complexity where it really isn't necessary. Also it seems like some settings are a special case that are not in the map while others are not, doesn't really add up.

Reply:
The way I was thinking about this is that any time we add anything to the namespace/struct that needs to be hot-reloaded, we'd need to update the SettingsSystem as well. One advantage of using the map is that we only need to update the namespace and add a corresponding entry to the map without caring about how the SettingsSystem updates individual values. Unfortunately disadvantage is that we need the extra SettingType enum to determine what kind of value is to be stored in the pointer.

If you think that's still too much complexity I'll go ahead and change things around!

Reply:
Yes we would need to add something if we add a configuration value. But if we add a configuration, we probably don't just need to change that anyways, somewhere something is going to use that value and that will need to be updated anyways.